src/ida_pro_mcp/mcp-plugin.py

```diff
@@ -13,16 +13,21 @@
     Any,
     Callable,
     get_type_hints,
-    TypedDict,
     Optional,
     Annotated,
     TypeVar,
     Generic,
-    NotRequired,
     overload,
     Literal,
 )
 
+# Use typing_extensions for TypedDict and NotRequired on Python < 3.12
+# to ensure compatibility with Pydantic
+if sys.version_info >= (3, 12):
+    from typing import TypedDict, NotRequired
+else:
+    from typing_extensions import TypedDict, NotRequired

```

Comment:
The `typing_extensions` package is not available inside of IDAPython per default, so this would conflict with the main use case. Such hacks should instead be in `idalib_server.py`